packages/core/src/browser/window/window-title-service.ts

```diff
@@ -0,0 +1,100 @@
+// *****************************************************************************
+// Copyright (C) 2022 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { Emitter, Event } from '../../common/event';
+import { PreferenceService } from '../preferences';
+
+export const WindowTitleParts = {
+    'activeEditorShort': true,
+    'activeEditorMedium': true,
+    'activeEditorLong': true,
+    'activeFolderShort': true,
+    'activeFolderMedium': true,
+    'activeFolderLong': true,
+    'folderName': true,
+    'folderPath': true,
+    'rootName': true,
+    'rootPath': true,
+    'appName': true,
+    'remoteName': true,
+    'dirty': true,
+    'developmentHost': true
+};
```

Comment:
I don't mean to make life difficult, but would it be possible to get the best of your original implementation (dynamic contributions) and support (most of) VSCode's options? I can imagine downstream IDE's being interested in adding contributions here, maybe variations on 'project type' fields or other custom status markers?

Reply:
I've made the `WindowTitleService` a bit more lenient in what it accepts. Downstream apps can now define custom parts 👍